### PR TITLE
Connection: Bring back auto-trigger for connection screen.

### PR DESCRIPTION
**Pull request by sergeymitr**
## Proposed changes:
Bring back connection auto-trigger for connection screen if `#setup` URL hash is used. Removed in #29443.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
pbNhbs-6l1-p2#comment-14837

## Does this pull request change what data or activity we track or use?
No.

## Testing instructions:
1. Disconnect Jetpack.
2. Load WP-admin Dashboard or Plugins page, you should see the "Set up Jetpack" banner.
3. Click on the banner, you should get redirected to the connection screen with `#setup` URL hash.
4. The connection screen flow should get triggered automatically: the button displays loader, you get redirected to Calypso connection flow shortly after.

### CODE REVIEW

**Review by jeherve — CHANGES_REQUESTED** [review]
You'll need to fix the failing checks before we can merge this, I'm afraid.

**Comment by sergeymitr** [comment]
Thanks @jeherve.
Fixed now, please give it another look when you can.
Since it's a critical component, and we're so far past the beta, I believe we should wait for the next release now.

**Review by jeherve — APPROVED** [review]

**Pull request merged**
